hash_table_implementation/hash_table.py

```diff
@@ -0,0 +1,86 @@
+class HashTable(object):
+
+
+    def __init__(self,size):
+        # Set up size and slots and data
+        self.size = size
+        self.slots = [None] * self.size
+        self.data = [None] * self.size
+
+    def put(self,key,data):
+        # Note, we'll only use integer keys for ease of use with the Hash Function
+        # Get the hash value
+        hashvalue = self.hashfunction(key,len(self.slots))
+
+        # If Slot is Empty
+        if self.slots[hashvalue] == None:
+            self.slots[hashvalue] = key
+            self.data[hashvalue] = data
+        else:
+            # If key already exists, replace old value
+            if self.slots[hashvalue] == key:
+                self.data[hashvalue] = data  
```

Comment:
Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `python spacing`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpc4mqt2xb/hash_table_implementation/hash_table.py
+++ b/tmp/tmpc4mqt2xb/hash_table_implementation/hash_table.py
@@ -19,7 +19,7 @@
         else:
             # If key already exists, replace old value
             if self.slots[hashvalue] == key:
-                self.data[hashvalue] = data  
+                self.data[hashvalue] = data
             # Otherwise, find the next available slot
             else:
                 nextslot = self.rehash(hashvalue,len(self.slots))
```